File: app/src/main/java/com/linxiao/quickdevframework/main/SimpleViewBindingFragment.java
package com.linxiao.quickdevframework.main;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.viewbinding.ViewBinding;

import com.linxiao.framework.architecture.BaseActivity;
import com.linxiao.framework.architecture.BaseFragment;

import java.lang.reflect.InvocationTargetException;
import java.lang.reflect.ParameterizedType;

/**
 * <p>
 * class usage summary
 * </p>
 *
 * @author linxiao
 * @since 2020-10-20
 */
public abstract class SimpleViewBindingFragment<B extends ViewBinding> extends BaseFragment {

    private B binding = null;

    protected B getViewBinding() {
        return binding;
    }

    @SuppressWarnings("unchecked")
    protected void setViewBinding(Class<B> bindingClass) {
        try {
            binding = (B) bindingClass.getMethod("inflate", LayoutInflater.class)
                    .invoke(null, getLayoutInflater());
        } catch (IllegalAccessException e) {
            e.printStackTrace();
        } catch (InvocationTargetException e) {
            e.printStackTrace();
        } catch (NoSuchMethodException e) {
            e.printStackTrace();
        }
    }

    @SuppressWarnings("unchecked")
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        setViewBinding((Class<B>) ((ParameterizedType) getClass()
                .getGenericSuperclass()).getActualTypeArguments()[0]);
        return binding.getRoot();
    }

}

File: framework/src/main/java/com/linxiao/framework/architecture/SimpleViewBindingFragment.kt
package com.linxiao.framework.architecture

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.viewbinding.ViewBinding
import java.lang.reflect.ParameterizedType

/**
 *
 *
 * class usage summary
 *
 *
 * @author linxiao
 * @since 2020-10-20
 */
@Suppress("UNCHECKED_CAST")
abstract class SimpleViewBindingFragment<B : ViewBinding?> : BaseFragment() {
    protected var viewBinding: B? = null
        private set

    protected fun setViewBinding(bindingClass: Class<B>) {
        try {
            viewBinding = bindingClass.getMethod("inflate", LayoutInflater::class.java)
                .invoke(null, layoutInflater) as B
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        setViewBinding((javaClass.genericSuperclass as ParameterizedType).actualTypeArguments[0] as Class<B>)
        return viewBinding!!.root
    }
}
